great_expectations/checkpoint/v1_checkpoint.py

```diff
@@ -0,0 +1,122 @@
+    validations: List[ValidationConfig]
+    actions: List[ValidationAction]
+    id: Union[str, None] = None
+
+    class Config:
+        arbitrary_types_allowed = (
+            True  # Necessary for compatibility with ValidationAction's Marshmallow dep
+        )
+        """
+        When serialized, the validations field should be encoded as a set of identifiers.
+        These will be used as foreign keys to retrieve the actual objects from the appropriate stores.
+
+        Example:
+        {
+            "name": "my_checkpoint",
+            "validations": [
+                {
+                    "name": "my_first_validation",
+                    "id": "a758816-64c8-46cb-8f7e-03c12cea1d67"
+                },
+                {
+                    "name": "my_second_validation",
+                    "id": "1339js16-64c8-46cb-8f7e-03c12cea1d67"
+                },
+            ],
+            "actions": [
+                {
+                    "name": "my_slack_action",
+                    "slack_webhook": "https://hooks.slack.com/services/ABC123/DEF456/XYZ789",
+                    "notify_on": "all",
+                    "notify_with": ["my_data_docs_site"],
+                    "renderer": {
+                        "class_name": "SlackRenderer",
+                    }
+                }
+        """
+        json_encoders = {
+            ValidationConfig: lambda v: v.serialize(),
+            Renderer: lambda r: r.serialize(),
+        }
+
+    @validator("validations", pre=True)
+    def _validate_validations(
+        cls, validations: list[ValidationConfig] | list[dict]
+    ) -> list[ValidationConfig]:
+        if len(validations) == 0:
+            raise ValueError("Checkpoint must contain at least one validation")
+
+        if isinstance(validations[0], dict):
+            validation_config_store = project_manager.get_validation_config_store()
+            identifier_bundles = [_IdentifierBundle(**v) for v in validations]  # type: ignore[arg-type] # All validation configs are dicts if the first one is
+            return cls._deserialize_identifier_bundles_to_validation_configs(
+                identifier_bundles=identifier_bundles, store=validation_config_store
+            )
+
+        return cast(List[ValidationConfig], validations)
+
+    @classmethod
+    def _deserialize_identifier_bundles_to_validation_configs(
+        cls, identifier_bundles: list[_IdentifierBundle], store: ValidationConfigStore
+    ) -> list[ValidationConfig]:
+        validations: list[ValidationConfig] = []
+        for id_bundle in identifier_bundles:
+            key = store.get_key(name=id_bundle.name, id=id_bundle.id)
+
+            try:
+                validation_config = store.get(key=key)
+            except (KeyError, gx_exceptions.InvalidKeyError):
+                raise ValueError(
+                    f"Unable to retrieve validation config {id_bundle} from store"
+                )
+
+            if validation_config:
```

Comment:
Do we need this `if`? With the try/except above, it looks like this will need to be defined.

Reply:
did it to appease `mypy` but updated to be a bit clearer